src/stream-ciphers.rst

```diff
@@ -255,83 +255,81 @@ Conclusion
 ^^^^^^^^^^
 
 In the real world, block ciphers are used in systems that encrypt large
-amounts of data all the time. We've seen that when using :term:`ECB mode`, an
-attacker can both analyze ciphertexts to recognize repeating patterns,
-and even decrypt messages when given access to an :term:`encryption oracle`.
+amounts of data all the time. We see that when using :term:`ECB mode`, an
+attacker both analyzes ciphertexts to recognize repeating patterns,
+and even decrypts messages when given access to an :term:`encryption oracle`.
 
 Even when we use idealized block ciphers with unrealistic properties,
-such as block sizes of more than a thousand bits, an attacker ends up
-being able to decrypt the ciphertexts. Real world block ciphers only
-have more limitations than our idealized examples, such as much smaller
+such as block sizes of more than a thousand bits, an attacker can
+decrypt the ciphertexts. Real world block ciphers
+have more limitations than our idealized examples, for example, having much smaller
 block sizes.
 
-We aren't even taking into account any potential weaknesses in the block
-cipher. It's not AES (or our test block ciphers) that cause this
-problem, it's our ECB construction. Clearly, we need something better.
+We are not yet even considering potential weaknesses in the block
+cipher. It is not AES nor the test block ciphers that cause the
+problem, it is our ECB construction. Clearly, something better is needed.
 
 Block cipher modes of operation
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-One of the more common ways of producing a :term:`stream cipher` is to use a
+A common way of producing a :term:`stream cipher` is by using a
 block cipher in a particular configuration. The compound system behaves
-like a :term:`stream cipher`. These configurations are commonly called
-:term:`mode of operation`\s. They aren't specific to a particular block cipher.
+like a :term:`stream cipher`. These configurations are known as
+:term:`mode of operation`\s. They are not specific to a particular block cipher.
 
-:term:`ECB mode`, which we've just seen, is the simplest such :term:`mode of operation`.
+:term:`ECB mode`, which we have just seen, is the simplest such :term:`mode of operation`.
 The letters ``ECB`` stand for electronic code book [#]_. For reasons
-we've already gone into, :term:`ECB mode` is very ineffective. Fortunately,
-there are plenty of other choices.
+we already discussed, :term:`ECB mode` is very ineffective. Fortunately,
```

Comment:
@lvh ,

I think "very ineffective" should be changed to "insecure"

Reply:
agreed

Reply:
Changed